src/Framework.RabbitMq.Consumer/Services/RabbitMqMultipleConsumersSemaphore.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System.Text.Json;
+
+using Framework.RabbitMq.Consumer.Interfaces;
+using Framework.RabbitMq.Consumer.Models;
+using Framework.RabbitMq.Consumer.Settings;
+
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+
+namespace Framework.RabbitMq.Consumer.Services;
+
+public class RabbitMqMultipleConsumersSemaphore(
```

Comment:
I think "custom lider election" it is not a good idea
Try to use exist variants ex https://kubernetes.io/docs/concepts/architecture/leases/

Reply:
Or Use BSS FW Named Lock Or MSSQL https://learn.microsoft.com/en-us/sql/relational-databases/system-stored-procedures/sp-getapplock-transact-sql?view=sql-server-ver16

Reply:
yes, object as lock was a bad idea. I changed it to fw named locks